images/kindnetd/cmd/kindnetd/main.go

```diff
@@ -54,10 +61,31 @@ func main() {
 	if err != nil {
 		panic(err.Error())
 	}
+
+	// override the internal apiserver endpoint to avoid
+	// waiting for kube-proxy to install the services rules.
+	// If the endpoint is not reachable, fallback the internal endpoint
+	controlPlaneEndpoint := os.Getenv("CONTROL_PLANE_ENDPOINT")
+	if controlPlaneEndpoint != "" {
```

Comment:
this block still seems wrong. if CONTROL_PLANE_ENDPOINT is set we should trust it, kindnetd is not for external usage.
this polling seems likely to wind up the buggy thing blocking startup more than anything.